### PR TITLE
[Merged by Bors] - feat: port LinearAlgebra.LinearPMap

### PR DESCRIPTION
**Pull request by mcdoll**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mcdoll** [comment]
blocked by leanprover/lean4#2074

File: Mathlib/LinearAlgebra/LinearPMap.lean [hunk 1]
```diff
@@ -0,0 +1,926 @@
...
+
+/-- A `LinearPMap R E F` or `E →ₗ.[R] F` is a linear map from a submodule of `E` to `F`. -/
+structure LinearPMap (R : Type u) [Ring R] (E : Type v) [AddCommGroup E] [Module R E] (F : Type w)
+  [AddCommGroup F] [Module R F] where
+  domain : Submodule R E
+  toFun : domain →ₗ[R] F
+#align linear_pmap LinearPMap
+
+-- mathport name: «expr →ₗ.[ ] »
+notation:25 E " →ₗ.[" R:25 "] " F:0 => LinearPMap R E F
```
Comment (dupuisf) [inline Mathlib/LinearAlgebra/LinearPMap.lean:52]:
```suggestion
notation:25 E " →ₗ.[" R:25 "] " F:0 => LinearPMap R E F
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Comment by mcdoll** [comment]
bors merge

**Pull request closed without merging**